mesonbuild/build.py

```diff
@@ -803,12 +803,13 @@ def get_outputs(self):
     def get_extra_args(self, language):
         return self.extra_args.get(language, [])
 
-    def get_dependencies(self):
+    def get_dependencies(self, exclude=[]):
         transitive_deps = []
         for t in itertools.chain(self.link_targets, self.link_whole_targets):
-            transitive_deps.append(t)
-            if isinstance(t, StaticLibrary):
-                transitive_deps += t.get_dependencies()
+            if t not in transitive_deps and t not in exclude:
+                transitive_deps.append(t)
+                if isinstance(t, StaticLibrary):
+                    transitive_deps += t.get_dependencies(transitive_deps + exclude)
```

Comment:
Nitpick: it's easier to read if this is written as:

```python
if t in transitive_deps or t in exclude:
    continue
transitive_deps.append(t)
...
```

Reply:
ok, thanks. Let me rework. My python is rather rusty.